calabash-cucumber/lib/calabash-cucumber/usage_tracker.rb

```diff
@@ -56,6 +57,21 @@ def post_usage_async
       private
 
       # @!visibility private
+      def preferences
+        @preferences ||= Calabash::Cucumber::Preferences.new
+      end
+
+      # @!visibility private
+      def user_id
+        @user_id ||= preferences.user_id
+      end
+
+      # @!visibility private
+      def allowed_to_track
+        @allowed_to_track ||= preferences.usage_tracking
+      end
+
+      # @!visibility private
       def self.track_usage?
         @@track_usage && !self.xtc?
```

Comment:
> `@@track_usage` vs `@allowed_to_track`

Specifically, `@@track_usage` is class variable that acts as a (class) global on/off switch.

`@allowed_to_track` is the category of things we are allowed to track; it is a memoized instance variable.

I think I have made two mistakes here:

* memoizing - this should be read from the .plist every time it is called
* naming - it sounds like a boolean

The memoization is a problem when it is paired with the memoization done in the proposed irb methods.